bindings/node/test/clientEncryption.test.js

```diff
@@ -815,4 +828,220 @@ describe('ClientEncryption', function () {
   it('should provide the libmongocrypt version', function () {
     expect(ClientEncryption.libmongocryptVersion).to.be.a('string');
   });
+
+  describe('createEncryptedCollection()', () => {
+    /** @type {InstanceType<ClientEncryption>} */
+    let clientEncryption;
+    const client = new MockClient();
+    let db;
+    const collectionName = 'secure';
+
+    beforeEach(async function () {
+      clientEncryption = new ClientEncryption(client, {
+        keyVaultNamespace: 'client.encryption',
+        kmsProviders: { local: { key: Buffer.alloc(96, 0) } }
+      });
+
+      db = client.db('createEncryptedCollectionDb');
+    });
+
+    afterEach(async () => {
+      sinon.restore();
+    });
+
+    context('validates input', () => {
+      it('throws if options are omitted', async () => {
+        const error = await clientEncryption
+          .createEncryptedCollection(db, collectionName)
+          .catch(error => error);
+        expect(error).to.be.instanceOf(TypeError);
+      });
+
+      it('throws if options.createCollectionOptions are omitted', async () => {
+        const error = await clientEncryption
+          .createEncryptedCollection(db, collectionName, {})
+          .catch(error => error);
+        expect(error).to.be.instanceOf(TypeError);
+      });
+
+      it('throws if options.createCollectionOptions.encryptedFields are omitted', async () => {
+        const error = await clientEncryption
+          .createEncryptedCollection(db, collectionName, { createCollectionOptions: {} })
+          .catch(error => error);
+        expect(error).to.be.instanceOf(TypeError);
+      });
+
+      it('throws if options.createDateKeyOptions.keyAltNames are provided', async () => {
+        const error = await clientEncryption
+          .createEncryptedCollection(db, collectionName, {
+            provider: 'aws',
+            createCollectionOptions: { encryptedFields: { fields: [] } },
+            createDataKeyOptions: { keyAltNames: ['blah!'] }
+          })
+          .catch(error => error);
+        expect(error).to.be.instanceOf(MongoCryptInvalidArgumentError, /keyAltNames/);
+      });
+    });
+
+    context('when input is not the correct type do not validate', () => {
+      it('when options.encryptedFields.fields is not an array no key generation is performed', async () => {
```

Comment:
the `context` block should describe the scenario, the `it` block should describe the assertion.  Both the context blocks in these tests and the it blocks contain both scenario and assertions.  

Here's an example of breaking up scenario / assertion:

```typescript
context('when options.encryptedFields.fields is not an array', () => {
  it('does not generate any encryption keys', () => { ... });
}); 
```

Reply:
broke these up